### DISCUSSION TITLE
Plugins to improve multiselection in Obsidian

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#86] [Offtopic]
I just found out about [Obsidian Multi Cursor plugin](https://github.com/ollyfg/obsidian-multi-cursor). It's not yet available in community plugins, and I haven't tried it yet, but it can be manually downloaded. Quote from [a community plugin list pull request](https://github.com/obsidianmd/obsidian-releases/pull/589):
> Simpler than advanced-cursors, this plugin brings some sublime text keybindings and allows click-to-deselect on existing selections.

And here is the above mentioned _advanced-cursors_ plugin: https://github.com/SkepticMystic/advanced-cursors . It's already in community plugins, it's a new plugin too.

I haven't tried these yet.

**Comment by Taitava** [comment]
I forgot to mention that I havent done so much testing how SC's `{{selection}}` variable works when an editor contains multiple selections. If I remember correctly, I tried it once and it seemed like `{{selection}}` only picked the first selection, but I'm not sure anymore as it was some time ago.

**Comment by FelipeRearden** [comment]
> I just found out about [Obsidian Multi Cursor plugin](https://github.com/ollyfg/obsidian-multi-cursor). 
> 
> > Simpler than advanced-cursors, this plugin brings some sublime text keybindings and allows click-to-deselect on existing selections.
 
I didn't know about this one @Taitava !!! Thanks for sharing :)


> And here is the above mentioned _advanced-cursors_ plugin: https://github.com/SkepticMystic/advanced-cursors . It's already in community plugins, it's a new plugin too.

Advanced Cursors is AMAZING I have been using since the first betas ! I only have good thing to say about Advanced Cursors 
- Take a look @Taitava it has a lot of operations not only for multiple cursors: you can create regex queries to navigate inside the current note.

**Comment by FelipeRearden** [comment]
> I forgot to mention that I havent done so much testing how SC's `{{selection}}` variable works when an editor contains multiple selections. If I remember correctly, I tried it once and it seemed like `{{selection}}` only picked the first selection, but I'm not sure anymore as it was some time ago.

I'm gonna try {{selection}} and {{clipboard}} to see what happens. I get back to you with the result :)

**Reply by FelipeRearden** [reply]
You are right @Taitava !

{{selection}} and {{clipboard}} only returns the first item of a multiple selection :)

**Reply by Taitava** [reply]
Do you mean that you copied to clipboard when you had multiple selections? And then `{{clipboard}}` only gave you the first selection? What if you paste into another program (what is a Notepad equivalent in Mac?), does it give you all the things you had selected, or just the first one?

**Reply by FelipeRearden** [reply]
> Do you mean that you copied to clipboard when you had multiple selections? And then `{{clipboard}}` only gave you the first selection? 


1- Advanced Cursors to Select Multiple Cursors
2- Copy (cmd+c)
3- click on a empty line to go back to single cursor
3- echo {{clipboard}} (output ignore)

Result on Obsidian: first item


> What if you paste into another program (what is a Notepad equivalent in Mac?), does it give you all the things you had selected, or just the first one?


Doing this into another program give all the things I selected :)


ANOTHER TEST

1- Advanced Cursors to Select Multiple Cursors
2- Copy (cmd+c)
3- click on a empty line to go back to single cursor
3- echo {{selection}} (output clipboard)

Result on Obsidian: first item

Result on Notepad: first item

**Reply by Taitava** [reply]
Ok, so it works the way I suspected.

**Reply by FelipeRearden** [reply]
Yeah ! 

Have a great day!
